### PR TITLE
feat: Support for login and registration via a browser custom tab

### PR DESCRIPTION
**Pull request by xitij2000**
This change adds support for logging in and registering a new account using the browser. This can be useful for cases where the only way to log into the instatance is via a custom third-party auth provider.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @xitij2000! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

File: auth/src/main/java/org/openedx/auth/data/repository/AuthRepository.kt [hunk 1]
```diff
@@ -60,12 +69,16 @@ class AuthRepository(
     }
 
     private suspend fun AuthResponse.processAuthResponse() {
+        Log.d("MMMMMMMM", error.toString())
```
Comment (yusuf-musleh) [inline auth/src/main/java/org/openedx/auth/data/repository/AuthRepository.kt:72]: Seems like some debug code here

File: auth/src/main/java/org/openedx/auth/data/repository/AuthRepository.kt [hunk 2]
```diff
@@ -60,12 +69,16 @@ class AuthRepository(
...
         if (error != null) {
             throw EdxError.UnknownException(error!!)
         }
         preferencesManager.accessToken = accessToken ?: ""
         preferencesManager.refreshToken = refreshToken ?: ""
         preferencesManager.accessTokenExpiresAt = getTokenExpiryTime()
+        Log.d("MMMMMMMM", preferencesManager.accessToken)
+        Log.d("MMMMMMMM", preferencesManager.refreshToken)
+        Log.d("MMMMMMMM", preferencesManager.accessTokenExpiresAt.toString())
```
Comment (yusuf-musleh) [inline auth/src/main/java/org/openedx/auth/data/repository/AuthRepository.kt:81]: Similarly some debugging code here

**Review by yusuf-musleh — APPROVED** [review]
@xitij2000 Great work on this! It's working quite well when I tested it out. I just had a small comment on some debug code pushed. I think it would also be helpful to add some documentation/instructions either in the README or the [Documentation.md](https://github.com/openedx/openedx-app-android/blob/main/Documentation/Documentation.md) about how to enable/setup the browser based authentication.

- [X] I tested this: (Configured and built the app on my phone with browser based auth with @xitij2000 server running the LMS/CMS, and tested registration/login flows)
- [X] I read through the code
- [ ] ~I checked for accessibility issues~
- [ ] Includes documentation: Missing but flagged it
- [ ] ~I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.~

**Review by yusuf-musleh — APPROVED** [review]

**Comment by xitij2000** [comment]
@yusuf-musleh I've updated the PR to remove the debugging logs. 
As for the documentation, I don't see much other relevant documentation in the readme or documentation.md, but will add a new document.

**Comment by yusuf-musleh** [comment]
@xitij2000 The new doc looks good to me! 👍

**Comment by mphilbrick211** [comment]
Hi @xitij2000! Just checking in on this.

**Comment by xitij2000** [comment]
> Hi @xitij2000! Just checking in on this.

I'd love some feedback on the approach etc. If that seems OK I can update the PR and get it ready to merge.

**Comment by marcotuts** [comment]
A few notes @xitij2000 on my end - This is good to go on the product review side, though I would suggest that this be set up as a configurable feature flag defaulted to being off. This will keep the native login / register as the default in the mobile applications. 

Additionally, there may be some need for conflict resolution and rebasing for the review to continue. @volodymyr-chekyrta  can review again once these items are addressed (and any other items as well that came up in the review above that may be outstanding. ) 

thanks!

**Comment by xitij2000** [comment]
> A few notes @xitij2000 on my end - This is good to go on the product review side, though I would suggest that this be set up as a configurable feature flag defaulted to being off. This will keep the native login / register as the default in the mobile applications.
> 
> Additionally, there may be some need for conflict resolution and rebasing for the review to continue. @volodymyr-chekyrta can review again once these items are addressed (and any other items as well that came up in the review above that may be outstanding. )
> 
> thanks!

Thanks for the review! The intention is definitely to keep the current login experience intact, and just allow this alternative flow for cases where it's needed.

**Comment by volodymyr-chekyrta** [comment]
@xitij2000, could you please resolve conflicts so I can proceed with review?

**Comment by xitij2000** [comment]
> @xitij2000, could you please resolve conflicts so I can proceed with review?

This PR is in the similar position as https://github.com/openedx/openedx-app-ios/pull/65#issuecomment-2048174354

I've tried rebasing but need more time that I have currently, since the rebased version is nto working.

**Comment by openedx-webhooks** [comment]
@xitij2000 Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-app-android&entry.867055334=https://github.com/openedx/openedx-app-android/pull/190&entry.1484655318=https://github.com/xitij2000&entry.752974735=2024-01-18+07:04&entry.1917517419=2024-05-21+10:54&entry.2133058324=No) so we can improve your experience in the future.

**Pull request closed without merging**